modules/core/src/main/java/org/apache/ignite/internal/management/cache/IdleVerifyResult.java

```diff
@@ -414,4 +379,212 @@ private void printConflicts(Consumer<String> printer) {
     @Override public String toString() {
         return S.toString(IdleVerifyResult.class, this);
     }
+
+    /** */
+    private static <K, V> Map<K, V> notNullFinalMap(@Nullable Map<K, V> map) {
+        return map == null ? Collections.emptyMap() : Collections.unmodifiableMap(map);
+    }
+
+    /** */
+    private static <V> List<V> notNullFinalList(@Nullable List<V> lst) {
+        return lst == null ? Collections.emptyList() : Collections.unmodifiableList(lst);
+    }
+
+    /**
+     * Builds result holding errors only.
+     *
+     * @see #builder()
+     */
+    public static IdleVerifyResult ofErrors(Map<ClusterNode, Exception> exceptions) {
+        return new IdleVerifyResult(exceptions);
+    }
+
+    /** @return A fresh result builder. */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /** Builder of {@link IdleVerifyResult}. Is not thread-safe. */
+    public static final class Builder {
+        /** */
+        private @Nullable Map<PartitionKey, List<PartitionHashRecord>> partHashes;
```

Comment:
Why do you need this intermediate collection? Let's fill all required collections incrementally in public Builder methods.